cpp/src/common/cumlHandle.cpp

```diff
@@ -189,9 +189,13 @@ cumlHandle_impl::cumlHandle_impl(int n_streams)
       return cur_dev;
     }()),
     _num_streams(n_streams),
+    _cublasInitialized(false),
+    _cusolverDnInitialized(false),
+    _cusparseInitialized(false),
     _deviceAllocator(std::make_shared<defaultDeviceAllocator>()),
     _hostAllocator(std::make_shared<defaultHostAllocator>()),
-    _userStream(NULL) {
+    _userStream(NULL),
+    _devicePropInitialized(false) {
```

Comment:
IMHO device properties is almost always needed so that should be initialized in the constructor.

Reply:
1. when I checked, `getDeviceProperties` call is not used anywhere else in cuML (currently).
2. when we last [measured](https://devblogs.nvidia.com/cuda-pro-tip-the-fast-way-to-query-device-properties/) the perf of `cudaGetDeviceProperties` it was one of the costly calls.

Hence, lazy-initialized it too.

Reply:
OK makes sense. Thanks for clarifying.